File: leetcode/linked-lists/237-delete-node-in-a-linked-list/c/solution.c
#include <stdlib.h>

/// {{
/// Problem: 237. Delete Node in a Linked List  
/// Difficulty: Medium  
/// Links: https://leetcode.com/problems/delete-node-in-a-linked-list  
/// Topics: `linked-list`  
/// Timestamp: `Thu, 19 Jun 2025 20:36:21 +0530`

// Definition for singly-linked list.
struct ListNode {
  int val;
  struct ListNode *next;
};

void deleteNode(struct ListNode *node) {
  struct ListNode *prev, *cur, *next;

  for (
    prev = NULL, cur = node;
    cur != NULL && cur->next != NULL;
    prev = cur, cur = cur->next
  ) {
    cur->val = cur->next->val;
  }

  if (prev != NULL) {
    prev->next = NULL;
  }
}

/// }}

File: leetcode/linked-lists/237-delete-node-in-a-linked-list/js/solution.js
/// {{
/// Problem: 237. Delete Node in a Linked List
/// Difficulty: `Medium`
/// Links: https://leetcode.com/problems/delete-node-in-a-linked-list
/// Topics: `linked-list`
/// Timestamp: `Mon, 11 Aug 2025 18:06:51 +0530`

// Definition for singly-linked list.
function ListNode(val) {
  this.val = val;
  this.next = null;
}

/**
 * @param {ListNode} node
 * @return {void} Do not return anything, modify node in-place instead.
 */
function deleteNode(node) {
  node.val = node.next.val;
  node.next = node.next.next;
}

/// }}

